libraries/utils/src/shared-components/drop-down.ts

```diff
@@ -1,66 +1,91 @@
     this._isMouseOverElement = false;
-    this._expanded = false;
-    this._updateElements(elements);
+    this._label = label;
+
+    this.isExpanded = false;
+    this.root = ui.div();
 
+    this._updateElement(elementCreation);
     this._initEventListeners();
   }
 
+
+  private _updateElement(elementCreation: () => HTMLElement) {
+    this._element = elementCreation();
+
+    this._dropDownElement = ui.div(ui.div(this._element), 'ui-drop-down-content');
+    this._dropDownElement.style.visibility = 'hidden';
+
+    this.root = ui.div([this._label, this._dropDownElement], 'ui-drop-down-root');
+  }
+
   private _initEventListeners() {
-    this._root!.addEventListener('mousedown', (e) => {
+    this.root.addEventListener('mousedown', (e) => {
       // check if the button is LMB
       if (e.button !== 0) return;
       if (this._isMouseOverElement) return;
 
-      if (!this._expanded)
-        this._expand();
-      else
-        this._collapse();
+      this._setExpandedState(this.isExpanded);
     });
 
-    this._dropDownElements!.addEventListener('mouseover', () => {
+    this._dropDownElement.addEventListener('mouseover', () => {
       this._isMouseOverElement = true;
     }, false);
 
-    this._dropDownElements!.addEventListener('mouseleave', () => {
+    this._dropDownElement.addEventListener('mouseleave', () => {
       this._isMouseOverElement = false;
     }, false);
-  }
 
-  private _updateElements(elements: string[]) {
-    this._elements = elements;
+    document.addEventListener('click', (event) => {
+      if (this.root.contains(event.target as Node)) return;
+      if (!this.isExpanded) return;
 
-    this._dropDownElements = ui.div(this._elements.map((elem) => ui.div(elem)), 'd4-combo-drop-down');
-    this._dropDownElements.style.visibility = 'hidden';
-
-    const dropDown = ui.div(this._dropDownElements, 'd4-combo-drop-down-fixed');
-    this._root = ui.div(['DropDown', dropDown], 'd4-combo-popup');
+      this._setExpandedState(this.isExpanded);
+    });
   }
 
-  private _expand() {
-    this._expanded = true;
-    this._root!.classList.add('d4-combo-popup-expanded');
-    this._dropDownElements!.style.visibility = 'visible';
-  }
+  private _setExpandedState(isExpanded: boolean) {
+    this.isExpanded = !isExpanded;
+    if (isExpanded) {
+      this.root.classList.remove('ui-drop-down-root-expanded');
+      this._dropDownElement.style.visibility = 'hidden';
+      return;
+    }
 
-  private _collapse() {
-    this._expanded = false;
-    this._root!.classList.remove('d4-combo-popup-expanded');
-    this._dropDownElements!.style.visibility = 'hidden';
+    this.root.classList.add('ui-drop-down-root-expanded');
+    this._dropDownElement.style.visibility = 'visible';
   }
 
 
-  get root() {
-    return this._root!;
+  get onExpand(): Observable<MouseEvent> {
+    return fromEvent<MouseEvent>(this.root, 'click')
+      .pipe(
+        filter(() => !this._isMouseOverElement)
+      );
+  }
```

Comment:
Such methods separation (each call on new line) is redundant. Consider the following:
```
get onExpand(): Observable<MouseEvent> {
  return fromEvent<MouseEvent>(this.root, 'click').pipe(filter(() => !this._isMouseOverElement));
}
```